### PR TITLE
lint fix for `flashers` module 

### PR DESCRIPTION
**Pull request by YifeiZuo01**
## Status
**READY**

## Migrations
NO

## Description
This is lint fix for mbed_flasher/flashers mudole.

### CODE REVIEW

File: mbed_flasher/flashers/FlasherMbed.py [hunk 1]
```diff
@@ -71,8 +85,14 @@ def reset_board(self, serial_port):
             else:
                 self.logger.info("reset failed")
         port.close()
-    
+
+
+    # no need to use self
+    # pylint: disable=no-self-use
     def auxiliary_drive_check(self, drive):
```
Comment (juhhov) [inline mbed_flasher/flashers/FlasherMbed.py]: could change this to static method if self is not needed.

Comment (YifeiZuo01) [inline mbed_flasher/flashers/FlasherMbed.py]: fixed

**Comment by jonikula** [comment]
Why is CI failing?

**Comment by YifeiZuo01** [comment]
The unittest tests are okay, I was trying to config pylint plugin into flasher, but somehow the archived file path seems not found by CI. I will keep look at it.

File: mbed_flasher/flashers/FlasherMbed.py [hunk 2]
```diff
@@ -128,6 +128,7 @@ def runner(self, drive):
                 self.logger.debug("re-mount check timed out for %s", drive[0])
                 break
 
+    # Todo: the code here might need improvement
     # pylint: disable=too-many-branches
     def check_points_unchanged(self, target):
         """
```
Comment (YifeiZuo01) [inline mbed_flasher/flashers/FlasherMbed.py]: @VeliMattiLahtela I changed the logic here a bit, could you please help verify it doesn't break anything?

File: mbed_flasher/flashers/enhancedserial.py [hunk 3]
```diff
@@ -11,12 +11,16 @@
...
 import re
+import termios
+from time import sleep
 import pkg_resources
 from serial import Serial, SerialException, SerialTimeoutException
-from time import sleep
 
 
-class EnhancedSerial(Serial):
+class EnhancedSerial(Serial): # pylint: disable=too-many-ancestors
```
Comment (jupe) [inline mbed_flasher/flashers/enhancedserial.py]: This is also class that could be common python lib repository, @juhhov @jonikula

**Comment by YifeiZuo01** [comment]
@VeliMattiLahtela could you please help check the refactor of flash() [here](https://github.com/ARMmbed/mbed-flasher/pull/54/commits/033183b6fed8d308441cd606196f2eb6bb5dc0c5) works?

**Comment by VeliMattiLahtela** [comment]
As the CI seems to be running as it has, there shouldn't be any breaking changes so LGTM

File: mbed_flasher/flashers/FlasherMbed.py [hunk 4]
```diff
@@ -166,28 +163,58 @@ def check_points_unchanged(self, target):
...
+        :param target: old target
+        :param new_target: new target
+        :return: if all is well None, otherwise error code
+        """
+        if not new_target:
+            return
+
+        if 'dev_point' not in new_target:
+            self.logger.error("Target %s is missing /dev/disk/by-id/ point", target['target_id'])
+            return -12
```
Comment (YifeiZuo01) [inline mbed_flasher/flashers/FlasherMbed.py]: this is check mount point, is it necessary to put 'dev_point' check here?

**Pull request merged**